File: PathoFroh/src/main/java/com/patho/main/repository/jpa/StainingPrototypeRepository.java
package com.patho.main.repository.jpa;

import com.patho.main.model.preset.StainingPrototype;
import com.patho.main.model.preset.StainingPrototypeType;
import com.patho.main.repository.jpa.custom.StainingPrototypeRepositoryCustom;

import java.util.List;

public interface StainingPrototypeRepository
        extends BaseRepository<StainingPrototype, Long>, StainingPrototypeRepositoryCustom {

    List<StainingPrototype> findAllByOrderByPriorityCountDesc();

    List<StainingPrototype> findAllByTypeOrderByPriorityCountDesc(StainingPrototypeType type);
}

File: src/main/kotlin/com/patho/main/repository/jpa/StainingPrototypeRepository.kt
package com.patho.main.repository.jpa

import com.patho.main.model.preset.StainingPrototype
import com.patho.main.model.preset.StainingPrototypeType
import com.patho.main.repository.jpa.custom.StainingPrototypeRepositoryCustom

interface StainingPrototypeRepository : BaseRepository<StainingPrototype, Long>, StainingPrototypeRepositoryCustom {

    fun findAllByOrderByPriorityCountDesc(): List<StainingPrototype>

    fun findAllByTypeOrderByPriorityCountDesc(type: StainingPrototypeType): List<StainingPrototype>

    fun findAllByOrderByIndexInListAsc() : List<StainingPrototype>
}